### DISCUSSION TITLE
Issues with Transaction History and Budget Display

### DISCUSSION THREAD

**Discussion by XNizel** [firefly-iii/firefly-iii#8947] [Firefly III]
First of all, I would like to thank you for your hard work and responsiveness to user messages. Your application has been extremely useful to me over the past eight months.

However, I have encountered a few minor issues recently that I hope you can help me resolve.

Transaction History Issue: For the past few weeks, the application seems to show a transaction history dating back to January 2023, even though my first transactions only date from August 2023. Could you help me understand why this is happening ? 

Incomplete Budget History Display: I have noticed that for some budgets, the period history on the right does not display all the months, even though there are transactions linked to these budgets each month. What might be causing this ? 

Histogram Display Issue: When viewing a budget over a long period, the application displays a weekly histogram rather than a monthly one, even though all my budgets are set to be monthly. Is there a way to ensure the histogram is displayed monthly, as it was initially ?

I appreciate your assistance with these issues and look forward to your response.

Thanks.

**Comment by JC5** [comment]
Heya,

Perhaps the issues can be resolved by browsing to `/flush` to clear your cache. Other issues are hard to diagnose without a little bit more details.

**Reply by XNizel** [reply]
Hello, 

Thank you for your prompt response. I’ll try to provide more details, but perhaps what I perceive as an issue is simply normal behavior.

At /budgets/show/12 for exemple
No period for March and April for exemple but i have transaction every month in this Budget
![image](https://github.com/firefly-iii/firefly-iii/assets/38247806/0a9ee602-3abf-4086-a136-49a1d9f7e345)


Then when i click on show all transaction I have this weekly histogram rather than a monthly one, even though all my budgets are set to be monthly
![image](https://github.com/firefly-iii/firefly-iii/assets/38247806/339851b9-7659-4f9a-8d7e-0d531e638b53)
It seems to me that at the beginning of my use, it was monthly, but I’m no longer sure. However, I do know that I can find this monthly graph in the reports if needed.


The last thing I noticed is the periods offered in the reports. 
I started using this wonderful tool in August 2023, but the reports suggest periods before in 2023. It’s not very problematic, but it does make me wonder if I made an error or if this is normal behavior. I exported all my transactions to verify, and there are no “erroneous” transactions before August 2023.
![image](https://github.com/firefly-iii/firefly-iii/assets/38247806/e0123dfa-77d4-4352-873b-82196132a465)

Thank you in advance

**Reply by JC5** [reply]
>  No period for March and April for exemple but i have transaction every month in this Budget

Mmm, I've not been able to replicate this yet, but I'll try.

> Then when i click on show all transaction I have this weekly histogram rather than a monthly one, even though all my budgets are set to be monthly

The period is determined based on the number of transactions, it will switch to monthly in a while. In a future release, this will be better.

> I started using this wonderful tool in August 2023, but the reports suggest periods before in 2023. It’s not very problematic, but it does make me wonder if I made an error or if this is normal behavior

This is normal, it makes the months always be in the same place no matter which year it is.

**Reply by XNizel** [reply]
Thank you for your reply !
